File: TD/TD7/Code/4ABC.cpp
/* 1. Fonction -> Calculer la somme d'une ligne */
/* 2. Fonction -> Calculer la somme d'une colonne */
/* 3. Fonction -> Calculer la somme de deux diagonales */
#include <iostream>

using namespace std;

/* Fonction 1 */

float sommeLigne(float tab[5][5], int n)
{
  int som;
  
  som = 0;
  
  for (int i = 0; i < 5; ++i) {
    som += tab[n][i];
  }
  
  return som;
}

/* Fonction 2 */

float sommeColonne(float tab[5][5], int n)
{
  int som;
  
  som = 0;
  
  for (int i = 0; i < 5; ++i) {
    som += tab[i][n];
  }
  
  return som;
}

/* Fonction 3 */

float sommeDiagonale(float tab[5][5], int n, int &som1)
{
  int som;
  
  som = 0;
  som1 = 0;
  
  for (int i = 0; i < 5; ++i) {
    som += tab[i][i];
    som1 += tab[i][4 - i];
  }
  
  return som;
}

/* Utilisation */

int main()
{
  int n;
  int som1; // Somme de la deuxième diagonale
  float tab[5][5] = { {1, 2, 3, 4, 5},
    {1, 2, 3, 4, 5},
    {1, 2, 3, 4, 5},
    {1, 2, 3, 4, 5},
    {1, 2, 3, 4, 5} };
  
  n = 3;
  
  // Somme des éléments de ligne
  cout << "La somme des éléments de cette ligne est "
  << sommeLigne(tab, n) << endl;
  
  // Somme des éléments de colonne
  cout << "La somme des éléments de cette colonne est "
  << sommeColonne(tab, n) << endl;
  
  //Somme des éléments des diagonales
  cout << "La somme des éléments de la première diagonale est "
  << sommeDiagonale(tab, n, som1) << " et de la deuxième diagonale est "
  << som1 << endl;
  
  return 0;
}

File: TD/TD7/Code/Python/4ABC.py
#!/usr/bin/python
# 1. Calculer la somme d'une ligne
# 2. Calculer la somme d'une colonne
# 3. Calculer la somme de deux diagonales


def somme_ligne(tab, n):
    som = 0

    for i in range(len(tab[0])):
        som += tab[n][i]

    return som


def somme_colonne(tab, n):
    som = 0

    for i in range(len(tab)):
        som += tab[i][n]

    return som


def somme_diagonale(tab):
    som = 0
    som1 = 0

    for i in range(len(tab)):
        som += tab[i][i]
        som1 += tab[i][4 - i]

    return som, som1


tab = [[2, 4, 6, 8, 10] for i in range(5)]

# La somme des éléments de la deuxième ligne
print('La somme des éléments de la ligne 2 est', somme_ligne(tab, 1))

# La somme des éléments de la toisième colonne
print('La somme des éléments de la 4 colonne est', somme_colonne(tab, 3))

# La somme des diagonales
som, som1 = somme_diagonale(tab)
print('La somme des éléments d\'une diagonale est',
      som, 'et celle de l\'autre est', som1)
